sql/hive/src/test/scala/org/apache/spark/sql/hive/CachedTableSuite.scala

```diff
@@ -361,4 +350,81 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
       assert(spark.sharedState.cacheManager.lookupCachedData(samePlan).isDefined)
     }
   }
+
+  test("SPARK-27248 refreshTable should recreate cache with same cache name and storage level") {
```

Comment:
Although the test case name is `~~~ and storage level`. It seems that we missed to check storage level check for Hive table. Could you add that consistentily too?